references/cli/addon.go

```diff
@@ -428,6 +428,70 @@ If you don't want the Helm component, just omit the three Chart-related paramete
 	return cmd
 }
 
+// NewAddonPushCommand pushes an addon dir/package to a ChartMuseum
+func NewAddonPushCommand(c common.Args) *cobra.Command {
+	p := &pkgaddon.PushCmd{}
+	cmd := &cobra.Command{
+		Use:   "push",
+		Short: "uploads an addon package to ChartMuseum",
+		Long: `Uploads an addon package to ChartMuseum.
+
+Two arguments are needed <addon directory/package> and <name/URL of ChartMuseum>.
+
+The first argument <addon directory/package> can be:
+	- your conventional addon directory (containing metadata.yaml). We will package it for you.
+	- packaged addon (.tgz) generated by 'vela addon package' command
+
+The second argument <name/URL of ChartMuseum> can be:
+	- registry name (helm type). You can add your ChartMuseum registry using 'vela addon registry add'.
+	- ChartMuseum URL, e.g. http://localhost:8080`,
+		Example: `# Push the addon in directory <your-addon> to a ChartMuseum registry named <localcm>
+$ vela addon push your-addon localcm
+
+# Push packaged addon mongo-1.0.0.tgz to a ChartMuseum registry at http://localhost:8080
+$ vela addon push mongo-1.0.0.tgz http://localhost:8080
+
+# Force push, overwriting existing ones
+$ vela addon push your-addon localcm -f`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 2 {
+				return fmt.Errorf("two arguments are needed: addon directory/package, name/URL of Chart repository")
+			}
+
+			c, err := c.GetClient()
+			if err != nil {
+				return err
+			}
+
+			p.Client = c
+			p.Out = cmd.OutOrStdout()
+			p.ChartName = args[0]
+			p.RepoName = args[1]
+			p.SetFieldsFromEnv()
+
+			return p.Push(context.Background())
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&p.ChartVersion, "version", "v", "", "Override chart version pre-push")
+	f.StringVarP(&p.AppVersion, "app-version", "a", "", "Override app version pre-push")
```

Comment:
```suggestion
	f.StringVarP(&p.AppVersion, "app-version", "a", "", "Override app version pre-push")
```

can we use addon version as a default

Reply:
It indeed uses the addon version by default. But the user can override it if he wants. This behavior is aligned with helm.

Reply:
ok, it's fine